specification/appconfiguration/data-plane/Microsoft.AppConfiguration/preview/2023-05-01-preview/appconfiguration.json

```diff
@@ -0,0 +1,2079 @@
+        },
+        "content_type": {
+          "type": "string",
+          "description": "The content type of the value stored within the key-value."
+        },
+        "value": {
+          "type": "string",
+          "description": "The value of the key-value."
+        },
+        "last_modified": {
+          "format": "date-time",
+          "type": "string",
+          "description": "A date representing the last time the key-value was modified."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "The tags of the key-value"
+        },
+        "locked": {
+          "type": "boolean",
+          "description": "Indicates whether the key-value is locked."
+        },
+        "etag": {
+          "type": "string",
+          "description": "A value representing the current state of the resource."
+        }
+      }
+    },
+    "Snapshot": {
+      "type": "object",
+      "required": [
+        "filters"
+      ],
+      "properties": {
+        "name": {
+          "description": "The name of the snapshot.",
+          "type": "string",
+          "readOnly": true
+        },
+        "status": {
+          "description": "The current status of the snapshot.",
+          "type": "string",
+          "enum": [
+            "provisioning",
+            "ready",
+            "archived",
+            "failed"
+          ],
+          "x-nullable": false,
+          "x-ms-enum": {
+            "name": "SnapshotStatus",
+            "modelAsString": true
+          },
+          "readOnly": true
+        },
+        "filters": {
+          "description": "A list of filters used to filter the key-values included in the snapshot.",
+          "type": "array",
+          "minItems": 1,
+          "maxItems": 3,
+          "items": {
+            "$ref": "#/definitions/KeyValueFilter"
+          }
+        },
+        "composition_type": {
+          "description": "The composition type describes how the key-values within the snapshot are composed. The 'key' composition type ensures there are no two key-values containing the same key. The 'key_label' composition type ensures there are no two key-values containing the same key and label.",
+          "type": "string",
+          "enum": [
+            "key",
+            "key_label"
+          ],
+          "x-nullable": false,
+          "x-ms-enum": {
+            "name": "CompositionType",
+            "modelAsString": true
+          }
+        },
+        "created": {
+          "description": "The time that the snapshot was created.",
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "expires": {
+          "description": "The time that the snapshot will expire.",
+          "format": "date-time",
+          "type": "string",
+          "x-nullable": true,
```

Comment:
You should avoid the use of `x-nullable: true`. Properties with no value should simply be omitted from the payload.

Reply:
@mikekistler 

I thought `x-nullable` was needed to denote that it was possible for the property to be omitted. Meaning, without `x-nullable`, there should always be a value for expires. 

You're saying we don't need `x-nullable` and we can still omit it from the response?

Reply:
That's correct. You can omit any value that is not "required". `x-nullable: true" means that [the property can be present but with the value "null"](https://github.com/Azure/autorest/tree/main/docs/extensions#x-nullable).

Reply:
To resolve this issue, should I just remove `"x-nullable": true` here?

Reply:
Yes, just remove `"x-nullable": true`.

Reply:
Updated, thanks